system/include/emscripten/em_asm.h

```diff
@@ -66,14 +69,33 @@ void emscripten_asm_const_async_on_main_thread(
 
 // We can use the generic selection C11 feature (that clang supports pre-C11
 // as an extension) to emulate function overloading in C.
-// All pointer types should go through the default case.
+//
+// All pointer types go through the default case, which means we have different
+// defaults in `__wasm64__` and `__wasm32__` modes.
+//
+// Because the default is different we also need to include more non-default
+// cases under `__wasm32__` since, for example, we don't want `short` to end up
```

Comment:
```suggestion
// cases under `__wasm32__` since, for example, we don't want `short` to end up
```
"under wasm32" should maybe be either "under wasm64" or "compared to under wasm32"?